package/scripts/preinstall

```diff
@@ -15,3 +15,27 @@ then
 else
   exit 0
 fi
+
+# Clean up subdirs so that there's a clean upgrade from previous brew versions
+subdirs=(
+  ".git"
+  "Library/Homebrew"
+)
+
+arch=$(/usr/bin/uname -m)
+
+if [[ ${arch} == "x86_64" ]]
+then
+  brew_dir="/usr/local/Homebrew"
+else
+  brew_dir="/opt/homebrew"
+fi
+
+for subdir in "${subdirs[@]}"
+do
+  if [[ -d "${brew_dir}/${subdir}" ]]
+  then
+    echo "Deleting ${brew_dir}/${subdir} for clean upgrade..."
+    /bin/rm -rf "${brew_dir:?}"/"${subdir}"
```

Comment:
Shouldn't do this unconditionally because it'll mean that anyone with e.g. contributions to Homebrew they haven't pushed yet will be destroyed.